### ISSUE TITLE
Use lsp_types

### ISSUE THREAD

**Issue by pinbraerts** [sudarshan-reddy/lsp-client-rs#1]
Hey, I stumbled across your project while looking for some lsp client implementation for testing my server. Do you consider using the [lsp_types](https://docs.rs/lsp-types/latest/lsp_types/#) crate instead of writing your own protocol implementation?

**Comment by sudarshan-reddy** [comment]
Good idea. Apologies. I was a bit busy these few weeks. I'll try to get this done this weekend.

**Comment by pinbraerts** [comment]
Actually, I found a testing suite in the lib I was using. So no need to hurry. This is just a suggestion in case you didn't know about the library. Maybe you could contribute there, if there is something missing
